### PR TITLE
PSR-1382 Upgrade to Scala 3

### PR DESCRIPTION
**Pull request by cyeniceri**

### CODE REVIEW

File: app/uk/gov/hmrc/pensionschemereturn/connectors/PsrConnector.scala [hunk 1]
```diff
@@ -81,8 +81,8 @@ class PsrConnector @Inject()(config: AppConfig, http: HttpClientV2, apiAuditUtil
     psaPspId: String,
     credentialRole: String,
     userName: String
-  )(
-    implicit headerCarrier: HeaderCarrier,
+  )(implicit
```
Comment (westwater) [inline app/uk/gov/hmrc/pensionschemereturn/connectors/PsrConnector.scala:84]: i thought with scala 3 they make you move to given / using syntax?

Comment (cyeniceri) [reply app/uk/gov/hmrc/pensionschemereturn/connectors/PsrConnector.scala:84]: I think we can use both style with scala-3. In future, older version might be deprecated but for now it's fine.

I converted to try new version in this file and it looks working :) 
app/uk/gov/hmrc/pensionschemereturn/models/response/PsrVersionsEtmpResponse.scala

**Review by westwater — APPROVED** [review]

File: .scalafix.conf [hunk 2]
```diff
@@ -1,19 +1,14 @@
...
-}
-
 OrganizeImports {
     groupedImports = Merge
     importsOrder = Keep
     importSelectorsOrder = Keep
     coalesceToWildcardImportThreshold = 3
     expandRelative = true
-    removeUnused = true
+    removeUnused = false
```
Comment (westwater) [inline .scalafix.conf:11]: why was this changed?

Comment (cyeniceri) [reply .scalafix.conf]: reverted this unintentional change

**Review by westwater — APPROVED** [review]
